play/src/pusher/services/MatrixProvider.ts

```diff
@@ -61,6 +97,274 @@ class MatrixProvider {
+        if (!this.roomAreaFolderID) {
+            console.error(new Error(`Failed to add room : ${roomID} to room area folder `));
+            return Promise.resolve(roomID);
+        }
+
+        const roomLinkContent = {
+            via: [MATRIX_DOMAIN],
+        };
+
+        return axios
+            .put(
+                `${MATRIX_API_URI}_matrix/client/r0/rooms/${this.roomAreaFolderID}/state/m.space.child/${roomID}`,
+                roomLinkContent,
+                {
+                    headers: {
+                        Authorization: `Bearer ${await this.getAccessToken()}`,
+                    },
+                }
+            )
+            .then((response) => {
+                if (response.status === 200) {
+                    return Promise.resolve(roomID);
+                } else {
+                    return Promise.reject(new Error(`Failed to add room : ${roomID} to room area folder `));
+                }
+            })
+            .catch((error) => {
+                return Promise.reject(new Error(`Failed to add room : ${roomID} to room area folder `));
+            });
+    }
+
+    deleteRoom(roomID: string): Promise<void> {
+        return this.kickAllUsersFromRoom(roomID)
+            .then(() => {
+                return this.kickUserFromRoom(ADMIN_CHAT_ID, roomID);
+            })
+            .catch((error) => {
+                console.error("Failed to delete room : " + roomID);
+                return Promise.reject(new Error("Failed to delete room : " + roomID));
+            });
+    }
+
+    async kickAllUsersFromRoom(roomID: string): Promise<void> {
+        return axios
+            .get(`${MATRIX_API_URI}_matrix/client/r0/rooms/${roomID}/members`, {
+                headers: {
+                    Authorization: `Bearer ${await this.getAccessToken()}`,
+                },
+            })
+            .then((response) => {
+                if (response.status !== 200) {
+                    throw new Error(`Failed to fetch members for room: ${roomID}`);
+                }
+
+                const kickMembersPromises = response.data.chunk.reduce(
+                    (acc: Promise<void>[], currentMember: { state_key: string; content: { membership: string } }) => {
+                        if (currentMember.state_key !== ADMIN_CHAT_ID && currentMember.content.membership !== "join") {
+                            acc.push(this.kickUserFromRoom(currentMember.state_key, roomID));
+                        }
+                        return acc;
+                    },
+                    []
+                );
+
+                return Promise.all(kickMembersPromises);
+            })
+            .then(() => {
+                return Promise.resolve();
+            })
+            .catch((error) => {
+                console.error("Failed to kick all user from room : " + roomID, error);
+                return Promise.reject(new Error("Failed to kick all user from room : " + roomID));
+            });
```

Comment:
Let the error bubble up. You will error log the error any way when you do the final catch.